packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx

```diff
@@ -106,17 +100,18 @@ const useDraftInboundLines = (itemId: string) => {
     setDraftLines([...draftLines, newLine]);
   };
 
-  const updateDraftLine = (
-    patch: Partial<DraftInboundLine> & { id: string }
-  ) => {
-    const batch = draftLines.find(line => line.id === patch.id);
-    if (batch) {
-      const newBatch = { ...batch, ...patch, isUpdated: true };
-      const index = draftLines.indexOf(batch);
-      draftLines[index] = newBatch;
-      setDraftLines([...draftLines]);
-    }
-  };
+  const updateDraftLine = React.useCallback(
+    (patch: Partial<DraftInboundLine> & { id: string }) => {
+      const batch = draftLines.find(line => line.id === patch.id);
+      if (batch) {
+        const newBatch = { ...batch, ...patch, isUpdated: true };
+        const index = draftLines.indexOf(batch);
+        draftLines[index] = newBatch;
+        setDraftLines([...draftLines]);
+      }
+    },
+    [draftLines, setDraftLines]
+  );
```

Comment:
TBH I'm flip flopping on `useCallback`/`useMemo`. When they first came out I thought 'why not just memoize all the things" and then I caused a bunch of bugs for myself and it made the code harder to read.

Now i'm flopping back to 'why not memoize all the things!' ... I have no middle ground?

Reply:
heh. I have noticed a few `useCallback`s appearing 😃 
Would be good to know what the gotchas are if we are heading back into memoland.